solutions/usaco-600.mdx

```diff
@@ -2,19 +2,105 @@
 id: usaco-600
 source: USACO Platinum 2016 January
 title: Fort Moo
-author: Kevin Sheng
+author: Kevin Sheng, Ryan Chou
 ---
 
-[Official Analysis](http://www.usaco.org/current/data/sol_fortmoo_platinum_jan16.html)
+[Official Analysis (Java)](http://www.usaco.org/current/data/sol_fortmoo_platinum_jan16.html)
 
+## Solution
+
+### Explanation
+
+By computing the [2D Prefix Sum](/silver/more-prefix-sums/) for the grid, we can evaluate the usability of a frame in $\mathcal{O}(1)$ time.
+
+Due to the low bounds on $N$, we can iterate through all possible pairs of columns and find the maximum area frame in $O(N)$ with a sliding window.
+
+### Implementation
+**Time Complexity:** $\mathcal{O}(N^3)$
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+int main() {
+	freopen("fortmoo.in", "r", stdin);
+	freopen("fortmoo.in", "w", stdout);
+
+	int n, m;
+	cin >> n >> m;
+
+	vector<vector<bool>> grid(n, vector<bool>(m, false));
+	vector<vector<int>> cs(n + 1, vector<int>(m + 1, 0));
+
+	for (int i = 0; i < n; i++) {
+		string s;
+		cin >> s;
+
+		for (int j = 0; j < m; j++) {
+			grid[i][j] = s[j] == 'X';
+
+			// build prefix sum
+			cs[i + 1][j + 1] = cs[i][j + 1] + cs[i + 1][j] - cs[i][j];
+			cs[i + 1][j + 1] += grid[i][j];
+		}
+	}
+
+	int ans = 0;
+
+	for (int c1 = 0; c1 < m; c1++) {
+		for (int c2 = 0; c2 < m; c2++) {
+			int prev = -1;
+
+			for (int r = 0; r < n; r++) {
+				bool emp = (cs[r + 1][c2 + 1] - cs[r][c2 + 1] - cs[r + 1][c1] + cs[r][c1]) == 0;
```

Comment:
80 chars?

Reply:
<img width="443" alt="image" src="https://user-images.githubusercontent.com/81596991/156711183-84767699-06a7-4830-b1fe-a47ec13bbc61.png">

Reply:
But does that include indentation?